frontend/javascripts/oxalis/model/reducers/settings_reducer.js

```diff
@@ -38,7 +40,29 @@ const updateActiveMapping = (
 function SettingsReducer(state: OxalisState, action: Action): OxalisState {
   switch (action.type) {
     case "UPDATE_USER_SETTING": {
-      const { propertyName, value } = action;
+      const { propertyName } = action;
+      let { value } = action;
+
+      switch (propertyName) {
```

Comment:
Can we access the JSON in `user_settings.schema.json` and just read `schema.definitions["types::UserSettings"].properties`? We could check whether `properties[propertyName].type === "number"` and if so read `.minimum` and `.maximum`? Then, we wouldn't need to define the limits twice?

If that's not possible, I'd still wonder whether we want to store the min/max values in a structured way so that the switch-block here can be replaced by a general clamping (i.e., `if (limits[propertyName]) { value = clamp(limits[propertyName].minimum, value, ...) } `).

Reply:
Thanks for pushing me to improve this code, I refactored the min/max handling according to our discussion :)

Reply:
It's always easier to push such things to others than on oneself :see_no_evil: